File: mo-auth/src/main/java/ink/ckx/mo/auth/exception/MyAuthenticationException.java
package ink.ckx.mo.auth.exception;

import ink.ckx.mo.common.core.result.ResultCode;
import lombok.Getter;
import org.springframework.security.core.AuthenticationException;

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/06
 */
@Getter
public class MyAuthenticationException extends AuthenticationException {

    private final ResultCode resultCode;

    public MyAuthenticationException(ResultCode resultCode) {
        super(resultCode.getMsg());
        this.resultCode = resultCode;
    }

    public MyAuthenticationException(ResultCode resultCode, String msg) {
        super(msg);
        this.resultCode = resultCode;
    }
}

File: mo-auth/src/main/kotlin/ink/ckx/mo/auth/exception/MyAuthenticationException.kt
package ink.ckx.mo.auth.exception

import ink.ckx.mo.common.core.result.ResultCode
import org.springframework.security.core.AuthenticationException

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/06
 */
class MyAuthenticationException : AuthenticationException {

    val resultCode: ResultCode

    constructor(resultCode: ResultCode) : super(resultCode.msg) {
        this.resultCode = resultCode
    }

    constructor(resultCode: ResultCode, msg: String?) : super(msg) {
        this.resultCode = resultCode
    }
}